ingest/gene_data_model.py

```diff
@@ -137,26 +110,33 @@ def chunk_gene_expression_documents(self):
         # storage size of the source file name and file type
         size_of_cell_names_field = 10 + 1  # "cell_names" is 10 characters
         size_of_field_expression_scores = 17 + 1
-        sum = 59 + len(self.source_file_name) + len(self.source_file_type)
+        starting_sum = 17 + len(self.source_file_type) + \
+            len(doc_name) + 1 + len(doc_path) + 1 + \
+            len(self.SUBCOLLECTION_NAME) + 1 + len(self.COLLECTION_NAME) + 1
         start_index = 0
         float_storage = 8
+        sum = starting_sum
 
         for index, cell_name in enumerate(self.cell_names):
 
-            cell_name_storage = len(cell_name) + 1 + size_of_cell_names_field
+            cell_name_storage = len(cell_name) + \
+                1 + size_of_cell_names_field
             expression_scores_storage = size_of_field_expression_scores + float_storage
             sum = sum + expression_scores_storage + cell_name_storage
-            # Subtract one and 32 based off of firestore storage guidelines for strings
+            # Subtract 32 based off of firestore storage guidelines for strings
             # and documents
             # This and other storage size calculation figures are derived from:
             # https://cloud.google.com/firestore/docs/storage-size
-            if (sum - 1 - 32) > DOCUMENT_LIMIT_BYTES:
-                end_index = index - 1
+            if (sum + 32) > DOCUMENT_LIMIT_BYTES or index == (len(self.cell_names) - 1):
+                if index == (len(self.cell_names) - 1):
+                    end_index = index
+                else:
+                    end_index = index - 1
+                print(sum)
```

Comment:
Unnecessary print statement here.

Reply:
I just have it there for debugging, making sure that it's chunking properly, and for other developers. It's easy to comment out if you're uninterested instead of figuring out where to print the sum.